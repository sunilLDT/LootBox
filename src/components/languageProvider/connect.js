/**
* Wrap component and pass props(strings, language)
* ```javascript
*   connect(SomeComponent);
* ```
*/

import  React, { Component } from 'react';
import PropTypes from 'prop-types';


const connect = (ComponentToWrap) => {
  return class LanguageComponent extends Component {
    constructor(p,c){
      super(p,c);
      this.update = () => this.forceUpdate();
    }
    static contextTypes = {
      strings: PropTypes.object.isRequired,
      language: PropTypes.object.isRequired
    }
    
    componentDidMount():void {
    // subscribe to future theme changes
      this.subscribe();
    }
    componentWillUnmount():void {
      // unsubscribe to changes
      this.unsubscribe();
    }
    render():React.Element<any> {
      const { strings, language } = this.context;
      return (
        <ComponentToWrap {...this.props} strings={strings} language={language}/>
      );
    }
    subscribe = () => {
      this.context.language.subscribe(this.update);
    }
    unsubscribe = () => {
      this.context.language.unsubscribe(this.update);
    }
  };
};
export default connect;
