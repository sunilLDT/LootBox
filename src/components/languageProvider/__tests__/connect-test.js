import { View } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import connect from '../connect.js';

it('renders correctly', () => {

  const TestComponent = connect(Test);
  const testFn = jest.fn();
  const options = {context: {strings: {}, language: {subscribe: testFn}}};
  const wrapper = shallow(<TestComponent/>, options);

  expect(wrapper).toMatchSnapshot();

  expect(wrapper.props()).toEqual(options.context);

  //mock component did mount behaviour
  wrapper.instance().subscribe();

  expect(testFn).toHaveBeenCalled();
});


class Test extends React.Component {
  constructor(props){
    super(props);
  }
  render(){
    return <View/>;
  }
}
