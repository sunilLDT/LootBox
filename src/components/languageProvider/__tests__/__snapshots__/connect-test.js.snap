exports[`test renders correctly 1`] = `
<Test
  language={
    Object {
      "subscribe": [Function],
    }
  }
  strings={Object {}} />
`;
